verify_room: the group check when enableDomainVerification is disabled

emrahcom opened the pull request:
The `room` doesn't check correctly when `enableDomainVerification` is disabled. The `group` is ignored and if there is a `group` then the `room` never matches.

This commit fixes this issue.

jitsi-jenkins commented:
Hi, thanks for your contribution!
If you haven't already done so, could you please make sure you sign our CLA (https://jitsi.org/icla for individuals and https://jitsi.org/ccla for corporations)? We would unfortunately be unable to merge your patch unless we have that piece :(.

emrahcom commented:
I added a second commit. This is mostly related in the code format on the changed block.
And I removed unnessary `?` char from regex. It works with it but no need.
Patch works on my side, I tested with different combinations

- room on link, room='*' on JWT
- room on link, room='matching-room' on JWT
- room on link, room='unmatching-room' on JWT (blocked)
- group+room on link, matching, unmatching and '*' for room on JWT (blocked only for unmatching room)
- group+room on link, matching, unmatching and '*' for room and maching group on JWT (blocked only for unmatching room)
- group+room on link, matching, unmatching and '*' for room and unmaching group on JWT (blocked for all cases)
- group+room on link, matching, unmatching and '*' for room and '*' from group on JWT
- room on link, matching, unmatching and '*' for room and a group on JWT (blocked)
- room on link, matching, unmatching and '*' for room and '*' for group on JWT

damencho commented:
Hey, @emrahcom the context information about user and group is just for informational purposes and we do not use it for checking.
I think I'm the source of the confusion, at least I remember mentioning it once or twice in the forum. But after you submitted the PR we revisited the docs about the token and we saw that tenant is not mentioned there. We will edit that doc with the correct information and will ping you with it.
Thank you for looking into this though, we still believe the current logic is correct after looking at it. 
To verify the group one needs to use the sub field to add the tenant there. And I will close the PR when update it with the link of the change we do in the doc.

damencho commented:
Here it is https://github.com/jitsi/lib-jitsi-meet/commit/a3e9f1c39f7f21d312429a13d783fe82951fb205

damencho commented:
Thank you, for bringing it up
